diff --git a/icicle/backend/cuda/cmake/Common.cmake b/icicle/backend/cuda/cmake/Common.cmake
index 88ba352..ebeac27 100644
--- a/icicle/backend/cuda/cmake/Common.cmake
+++ b/icicle/backend/cuda/cmake/Common.cmake
@@ -53,7 +53,7 @@ function(set_gpu_env)
     # Check CUDA version and, if possible, enable multi-threaded compilation 
     if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL "12.2")
         message(STATUS "Using multi-threaded CUDA compilation.")
-        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --split-compile 0" PARENT_SCOPE)
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --split-compile 1" PARENT_SCOPE)
     else()
         message(STATUS "Can't use multi-threaded CUDA compilation.")
     endif()
